unity/Assets/Scripts/InstantiatePrefabTest.cs

```diff
@@ -304,7 +304,7 @@ bool AlwaysPlaceUpright
 
                 Vector3 Offset = oabb.ClosestPoint(oabb.transform.TransformPoint(oabb.center) + -rsp.ReceptacleBox.transform.up * 10); // was using rsp.point
                 BoxBottom = new Plane(rsp.ReceptacleBox.transform.up, Offset);
-                DistanceFromBoxBottomTosop = BoxBottom.GetDistanceToPoint(sop.transform.position);
+                DistanceFromBoxBottomTosop = Math.Abs(BoxBottom.GetDistanceToPoint(sop.transform.position));
```

Comment:
I haven't read this super closely but are we sure we want the absolute value here? Just glancing at how this distance is used below it looks like it might need to be negative in some cases.

Reply:
I think so. This whole section is quite messy, but basically we are trying to get the distance from the bottommost point of the object bounding box to the actual object's transform.position, and use this as an offset in the `y` direction so that the object is placed "on top" of a receptacle. The "on top" is relative to whatever the receptacle's local `y` direction is, so the absolute value should still be fine even if a receptacle was upside down, for example.

Reply:
Gotcha, sounds good!